README.md

```diff
@@ -51,3 +72,172 @@ using the following command:
+
+Create the `local` and `remote` clusters and install tekton latest on them.
+Use two different terminals for that - one for each cluster.
+
+#### `local` cluster terminal
+```
+$ mkdir ~/.kube
+$ export KUBECONFIG=~/.kube/local
+$ kind create cluster --name=local
+$ kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
+```
+#### `remote` cluster terminal
+```
+$ export KUBECONFIG=~/.kube/remote
+$ kind create cluster --name=remote
+$ kubectl apply --filename https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
+```
+### Clone and run the operator
+```
+$ git clone git@github.com:redhat-appstudio/internal-services.git
+Cloning into 'internal-services'...
+```
+#### Installing the internal-services CR
+
+Both clusters should have the CRs installed, so the following steps should be done in both terminals, local and remote.
+```
+$ cd internal-services
+internal-services $ make manifests generate
+internal-services $ make install
+```
+
+#### Running the operator in the cluster
+
+The operator needs to run a single cluster - in this test case, the `local` cluster/terminal - and it
+will listen for `InternalRequests` created in the `remote` cluster denoted by the `--remote-cluster-config-file` parameter. 
+
+It also requires the configuration set in a `InternalServicesConfig` resource - still in the local cluster - with the following content:
+
+```
+cat > config.yaml  <<EOF
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: InternalServicesConfig
+metadata:
+  name: "config"
+spec:
+  allowList:
+    - default
+  debug: true
+  volumeClaim:
+    name: workspace
+    size: 1Gi
+EOF
+$ kubectl create -f config.yaml
+```
+Note that it *MUST* be called `config`.
+
+Then you can start the operator:
+
+
+```
+internal-services $ go run main.go --remote-cluster-config-file=$HOME/.kube/remote
+```
+
+### Testing an InternalRequest
+
+To process a `InternalRequest` the `internal-services-controller` requires a Pipeline created in the Cluster that the request should run in.
+
+#### Sample Pipeline
+
+In this example the following `sample` Pipeline should exist in the `local` cluster.
+```
+$ cat > pipeline.yaml <<EOF
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: sample
+spec:
+  tasks:
+  - name: sample
+    taskSpec:
+      results:
+      - name: output
+        description: just an example
+      steps:
+      - image: ubuntu
+        script:
+          echo "Output from today's demo" | tee $(results.output.path)
+  results:
+  - name: foo
+    description: result from the task
+    value: $(tasks.sample.results.output)
+EOF
+$ kubectl create -f pipeline.yaml
+
+```
+#### Sample InternalRequest
+
+An `InternalRequest` should be created in the `remote` cluster so the `internal-services-controller` can read and process it from the `local` cluster.
+The operator then creates a `PipelineRun` out of the defined `Pipeline` in the `InternalRequest` parameter `request`.
+
```

Comment:
It would be great if we add a note here saying InternalRequest must be created in the namespace mentioned in the `allowList:` of internalServiceConfig. Otherwise, the controller will not allow the InternalRequest.

Reply:
I think in case of `Kind` this should not be the case. The `Links` section goes to the doc. where this info is already explained.

Reply:
Ok. I can add. I was just wondering that if the person need more info it should follow the doc listed in the Links section. This one should be something "as is".

Reply:
@happybhati take a look again, pls!